reviews/models/mixins.py

```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from transitions import Machine
+
+from django.db import models
+
+from reviews.utils import AttrProxy
+from reviews.workflow import States
+from reviews.workflow import Workflows
+from reviews.workflow import TRANSITIONS
+
+
+class ReviewProviderSettings(models.Model):
+    """A reviewed/moderated collection of objects.
+    """
+
+    # TODO GenericForeignKey, if/when we have multiple types of providers
+    provider = models.OneToOneField('osf.PreprintProvider', related_name='reviews_settings', on_delete=models.CASCADE)
+
+    workflow = models.CharField(max_length=15, choices=Workflows.choices(), default=Workflows.PRE_MODERATION.value)
+    comments_public = models.BooleanField(default=False)
+    comments_anonymous = models.BooleanField(default=True)
+
+
+class ReviewableMixin(models.Model):
+    """Something that may be included in a reviewed collection and is subject to a reviews workflow
+    """
+
+    reviews_state = models.CharField(max_length=15, db_index=True, choices=States.choices(), default=States.PENDING.value)
+
+    def __init__(self, *args, **kwargs):
+        super(ReviewableMixin, self).__init__(*args, **kwargs)
+
+        self.__machine = Machine(
+            model=AttrProxy(self, state='reviews_state'),
```

Comment:
You make a good point, transitions gets pretty handsy with the model, which is bad for a mixin. It'd be more maintainable/readable to have actual method definitions instead of a bunch of magically added ones. I'll do that real quick.